packages/components/text-input/src/text-input.ts

```diff
@@ -19,33 +14,29 @@ let nextUniqueId = 0;
 /**
  * Single line text input component.
  *
+ * @csspart wrapper - The input's wrapper
  * @slot prefix - Content shown before the input
  * @slot input - The slot for the input element
  * @slot suffix - Content shown after the input
+ * @slot error-text - The error text to display
+ * @slot hint-text - The hint text to display
  */
-export class TextInput extends FormControlMixin(HintMixin(LitElement)) {
+export class TextInput extends FormControlMixin(ScopedElementsMixin(LitElement)) {
   /** @private */
-  static override styles: CSSResultGroup = [validationStyles, hintStyles, styles];
-
-  #onKeydown = (event: Event): void => {
-    if ((event as KeyboardEvent).key === 'Enter') {
-      this.input.form?.requestSubmit();
-    }
-  };
-  #events = new EventsController(this, {
-    click: this.#onClick
-  });
+  static get scopedElements(): ScopedElementsMap {
+    return {
+      'sl-error': Error,
+      'sl-hint': Hint,
+      'sl-icon': Icon
```

Comment:
We should also have added these components to the text-input's `package.json` file as dependencies?